src/screens/planItemActivity/PlanItemForm.tsx

```diff
@@ -0,0 +1,166 @@
+import { Formik, FormikProps } from 'formik';
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+import * as Yup from 'yup';
+
+import { Card, IconButton, IconToggleButton, StyledText, TextInput } from 'components';
+import { i18n } from 'locale';
+import { PlanItem, PlanItemType } from 'models';
+import { dimensions, getElevation, palette, typography } from 'styles';
+import { PlanItemComplexTask } from './PlanItemComplexTask';
+import { PlanItemSimpleTask } from './PlanItemSimpleTask';
+
+export interface PlanItemFormData {
+  name: string;
+  nameForChild: string;
+}
+
+interface Props {
+  onSubmit: (formData: PlanItemFormData) => Promise<void>;
+  planItem: PlanItem;
+}
+
+interface State {
+  taskType: PlanItemType;
+}
+
+export class PlanItemForm extends React.PureComponent<Props, State> {
+  state: State = {
+    taskType: this.props.planItem ? this.props.planItem.type : PlanItemType.SimpleTask,
+  };
+
+  initialValues: PlanItemFormData = {
+    name: this.props.planItem ? this.props.planItem.name : '',
+    nameForChild: this.props.planItem ? this.props.planItem.nameForChild : '',
+  };
+
+  validationSchema = Yup.object().shape({
+    name: Yup.string().required('Required!'),
+    nameForChild: Yup.string(),
+  });
+
+  isSimpleTask = (): boolean => {
+    const { planItem } = this.props;
+
+    if (!planItem) {
+      return true;
+    }
+    return planItem.type === PlanItemType.SimpleTask;
+  };
+
+  changePlanItemType = async (isSimpleTask: boolean) => {
```

Comment:
Task type toggling should be also extracted to the `PlanItem` model, similar as current `complete` method.